src/routes/concepts/components/event-handlers.mdx

```diff
@@ -3,68 +3,199 @@ title: Event Handlers
 
-Supported events such as `click`, `mouseover`, and `keydown` are just a few examples that are optimized in this way.
-To view the full list see the [references below](#delegated-events).
+Supported events such as `click`, `input` and `keydown` are just a few examples that are optimized in this way.
+To view the full list see the [references below](#list-of-delegated-events).
 
-If you need to attach an event listener to an element that is not supported by Solid's event delegation, you can use the [on:](/reference/jsx-attributes/on-and-oncapture) prefix followed by the event name, and assign it a function.
+If you need to attach an event listener to an element that is not supported by Solid's event delegation, such as a custom event in a [custom element](https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_custom_elements), you can use the [`on:__`](/reference/jsx-attributes/on-and-oncapture) form.
 
 ```tsx
 <div on:customEvent={handleCustomEvent} />
 ```
 
-## Binding events
+### Event delegation considerations
+
+While delegated events provide some performance enhancements, there are tradeoffs.
+
+Event delegation is designed for event propagation through the JSX Tree, rather than the DOM Tree.
+This can differ from the previous expectations of how events work and flow.
 
-In Solid, event handlers can be bound by passing an array to the event handler.
-The second item of the array is supplied as the handler's first argument.
+Some things to keep in mind include:
+
+- Delegated event listeners are added _once_ per event type and handle all future events of that type. This means that delegated event listeners remain active even if the element that added them and its handler is removed.
+
+  For example, if a `div` listens for `mousemove` and is later removed, the `mousemove` events will still be dispatched to the `document` in case a different element is also listening for mouse moves.
 
 ```tsx
-const handler = (data, event) => {
-	console.log("Data:", data, "Event:", event);
-};
+<div onMouseMove={handleCustomEvent} />
+```
 
-<button onClick={[handler, "Hello!"]}>Click Me</button>;
+<Callout type="tip" title="Ocassional Events">
+
+Rather than using delegated events for events that happen infrequently, **native events** are a better solution.
+Since these events happen in specific circumstances, they do not benefit from the performance improvements of event delegation.
+
+```tsx
+<div on:mousemove={handleCustomEvent} />
+```
+
+</Callout>
+
+- `event.stopPropogation()` does not work as expected since events are attached to the `document` rather than the `element`.
+
+With cases like this, a native event is recommended.
+As an example, using a native event would stop the following event from reaching the `div native` handler, which is _not_ the case for delegated events.
+You can [view this example in the Solid Playground](https://playground.solidjs.com/anonymous/c5346f84-01e4-4080-8ace-4443ffd0bb10).
```

Comment:
In the site it's formatted as if it's unrelated - I added some spaces because I think it will indent it ??? Before I merge I need to double check this one

```suggestion
     With cases like this, a native event is recommended.
     As an example, using a native event would stop the following event from reaching the `div native` handler, which is _not_ the case for delegated events.
     You can [view this example in the Solid Playground](https://playground.solidjs.com/anonymous/c5346f84-01e4-4080-8ace-4443ffd0bb10).
```